### ISSUE TITLE
Notification: screen is turned on and content is displayed

### ISSUE THREAD

**Issue by JPFrancoia** [Freeyourgadget/Gadgetbridge#1607]
#### Before requesting a new feature, please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.

#### Your issue is:
TL;DR: Notifications turn screen on and display content (text).

Long version:

On my phone lock screen's, notifications contents are not displayed. i.e, for whatsapp, I see the number of notifications on the lockscreen, but the content of the whatsapp messages can't be displayed on the lock screen. The phone needs to be unlocked first.

This is set in the notification settings menu of the phone.

However, on my mi band 4, with gadget bridge stock settings, when I receive a whatsapp message, the screen of the watch turns on and displays the content of the text message. Same goes for email. This is a bit annoying from a privacy point of view: people could be reading private messages.

I would like to know if there is a way to do any of the following:

- not turn screen on when a notification is received
- hide the content of the notification being received

Someone apparently has the oppposite problem: https://github.com/Freeyourgadget/Gadgetbridge/issues/1204

#### Your wearable device is:
 Mi band 4

#### Your android version is:

Custom ROM

#### Your Gadgetbridge version is:

0.35.2

**Comment by ashimokawa** [comment]
@JPFrancoia 
The screen will always turn on when you receive a message, I don't think the mi band firmware allows to change that.

So you do not want to see ANY message content at all on your Mi Band?

**Comment by JPFrancoia** [comment]
> The screen will always turn on when you receive a message, I don't think the mi band firmware allows to change that.

Yeah I kinda suspected that...

> So you do not want to see ANY message content at all on your Mi Band?

Ideally I could be able to select this behaviour on a per-app basis. I don't care if I see Twitter messages on the watch, but I do care if it's whatsapp messages. If this is not possible, I would prefer to not see ANY message content at all on the MiBand.

**Comment by IzzySoft** [comment]
So why not blacklist those you don't want to see? See [our wiki](https://codeberg.org/Freeyourgadget/Gadgetbridge/wiki/Configuration) for details on how to do that (watch out for "blacklist" in screenshots and text). Then you e.g. won't see any WA notifications at all. And that works on a per-app basis.

**Comment by JPFrancoia** [comment]
@IzzySoft yes, I know I can do that. I forgot yo mention in my previous message I still would like to see the number of notifications for whatsapp. I edited my message.

**Comment by IzzySoft** [comment]
Ah – yes, that was exactly what I wondered about. In this case I'm afraid there (currently) is no solution.

**Comment by JPFrancoia** [comment]
ok no problem, I tried at least. Thanks for your answer.

**Comment by ashimokawa** [comment]
I reopend this because it is of course possible to implement that.

**Comment by roshavagarga** [comment]
My 2 cents on the matter is that there are a few ways to implement this stylistically.
One way is to offer a general icon+text combination that says 'New notifications received'. The other option would be to have that same icon with the name of the app underneath.

Generally I think both should be available (if implementable). For instance, some people might not want others to see that they received a message on a certain application, while others may be fine with it.

**Comment by sunsero** [comment]
Alternatively you could disable all notifications except for Tasker in Gadgetbridge, intercept notifications of all apps with Tasker and then decide in a Tasker task which notification you want on the Mi Band and repeat the original notification with or without content as a Tasker notification to let Gadgetbridge send it to the Mi Band.

**Comment by roshavagarga** [comment]
@sunsero I'd rather not have to install an additional thing just to control something like notifications :)

**Comment by sunsero** [comment]
It would be the most flexible way however to send or to not to send content to mi band 4 - depending on generating app, sender, content, time, location, etc. Just my 2 cents ;)

**Comment by roshavagarga** [comment]
Flexibility doesn't matter in this case @sunsero 

Think of it like this, first of all GB is for all devices, not just the mb4. Hence, it would make more sense to:
1. Handle functions internally when possible, in a way that works on all devices that can support said function, then add customization or additional features to said functions per device.
2. Add the ability to make use of Tasker (or Tasker can do it? Depends on what needs to be fed into what I guess.)

The first part of point 1 is required, the second part of it is a good thing to have but depends on willingness and how hard it is to do. Point 2 is entirely optional and would be cool, but a core functionality shouldn't depend on an external app if that functionality can be coded into GB itself :) Some stuff can't be coded into GB, or they make use of open-source code to add it, since it's already been done by somebody - notifications shouldn't be something of this sort in my view.

If they can be handled via GB, they should be entirely handled by GB and implementing a Tasker connection should be something people should opt into, rather than be forced into. I realize you view Tasker as something nice to have, and I agree, it has its pluses, but I feel like most people wouldn't use it, hence forcing them to do so seems bad design, unless there's no other option on the table.

**Comment by sunsero** [comment]
Yes, I agree basic functionality should be provided by GB itself. Nicest way to implement an option to hide content would probably be in the blacklist for notifications. I don't think it has to be device specific since blacklist is also a global functionality and hiding content is something GB must do, not the device after receiving the notification.

Regarding Tasker: What I've described should already be doable without changing GB. Would probably be easier however if Tasker could send messages with the help of a Tasker plugin GB provides. Other useful functionalities of this plugin could be e.g. setting alarms, heartrate events or button press events (or swipes from the music screen in the case of MB4).
